### PR TITLE
Vue: Fix ugly formatting for one line template in SFC

### LINKED ISSUE #8151: Vue: ugly formatting for one line `template` in Vue SFC
**Prettier 2.0.5**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeeBbADgGwIbwB8qWhAZhBKgPSmoAmAlgG7mU1Os2a4FyEAdKCAA0ICFhiNoAZ2Sg8AJ0UQA7gAUlCOSjw5VeAJ5yxAI0V4wAazgwAylkuMoAc2QxFAVzhiAFjAwcAHVfRngZRzA4O20wljDDZHAZExBnGThFGHULFww8ZDI9DLEAKxkADwAhC2tbOzwMOAAZZzhC4p8Qcoq7ZxccOABFTwh4DpwSkEdFDMUk5m9RacVnGCDGehhfZAAOAAYxLBUMoIssJOO4OeZ2sQBHUfgciR0QPBkAWig4OHo-5aKOCPRhAnJ4PIFJBFSZdDIYRjuLxw-qDEZjdrQzpiGB4UwbLY7JAAJhxFkYOH6AGEIBh8klrgBWZaeDIAFTxOhhU0WcAAklB-rA7GBVpIAIKCuwwQyDCYZAC+CqAA)
```sh
--parser vue
```

**Input:**
```vue
<template><p>foo</p><div>foo</div></template>

```

**Output:**
```vue
<template
  ><p>foo</p>
  <div>foo</div></template
>

```

**Expected behavior:**

```vue
<template>
  <p>foo</p>
  <div>foo</div>
</template>
```

### PR DESCRIPTION
**Pull request by sosukesuzuki**
Fixes #8151

- [x] I’ve added tests to confirm my change works.
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/language-html/utils.js [hunk 1]
```diff
@@ -530,7 +530,15 @@ function getNodeCssStyleDisplay(node, options) {
...
     case "ignore":
       return "block";
-    default:
+    default: {
+      // See https://github.com/prettier/prettier/issues/8151
+      if (
+        options.parser === "vue" &&
+        node.name === "template" &&
+        node.parent.type === "root"
+      ) {
```
Comment (fisker) [inline src/language-html/utils.js:539]: Do we need check `lang`?

Comment (sosukesuzuki) [reply src/language-html/utils.js:539]: I think we don't need check `lang`. I'm not sure why it is needed.

**Comment by fisker** [comment]
Do  you  guys  think  this  related? Should we treat every root block as `block`?

**Prettier pr-8325**
[Playground link](https://deploy-preview-8325--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuEAeAJhMBnABAGwEMoBzAXgFcoBrKCAdygD5UB6THFjvI0ymuoxbss2JgB0oIADQgIABxgBLaNmShCAJ00MACloRqUhfPUIBPNbIBGmwmGpwYAZXn2lpZDE0U4suAC21nDo6CEAMsQkFIQkcABiEJoBhDDKnsYUMBAyIAAWMAH4AOp5SvDYbmBwzoblSgBu5ebI4NhWIB7YcJowunYkKcgAZibdsgBW2AAeAEJ2Dk7OhAFw4R5wI2N+IFPTzh4k+HAAihQQ8Fv44yBumt2arQ2+ufKaHjDFSugwecgAHAAGWRvCDdYp2eStN5wB4NTayACO53g-QURhAhGwAFooHAQiFcpo4MilMT+rEhkhRtcdt0AkovD46YdjmcLptqdtZDBCNYvj8-kgAEw8uxKfCHADCEACVJAsIArLkKN0ACp8ow0m7POAASSgYVgzjA70UAEFDc4YOZjldugBfB1AA)
```sh
--parser vue
```

**Input:**
```vue
<docs lang=unknown></docs><docs lang=unknown></docs>

```

**Output:**
```vue
<docs lang="unknown"></docs>><docs lang="unknown"></docs>

```

File: src/language-html/utils.js (hunk 1 above)
Comment (fisker) [reply src/language-html/utils.js:539]: Ya, seems we don't need that.

**Comment by thorn0** [comment]
> Should we treat every root block as block?

Sounds like the right thing to do for SFCs.

> Or just wrong solution in #8280

Treating root elements as blocks would solve that, right?

**Pull request merged**